src/utils.c

```diff
@@ -2414,6 +2419,48 @@ void utils_start_new_geany_instance(const gchar *doc_path)
 }
 
 
+
+static int get_path_max(const char *path)
+{
+#ifdef PATH_MAX
+	return PATH_MAX;
+#else
+	int path_max = pathconf(path, _PC_PATH_MAX);
+	if (path_max <= 0)
+		path_max = 4096;
+	return path_max;
+#endif
```

Comment:
I think the `pathconf` code should be the default, and `PATH_MAX` should only be considered on Windows. On Unix, `PATH_MAX` is not necessarily a constant.

Looks like the code from `realpath(3)` man page and it discusses problems with that code.

Apparently we should be doing `realpath(name, NULL)` but I'm unsure if it's supported on our relevant platforms?

Reply:
I found two different manpages online that may apply to Mac... [iOS](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/realpath.3.html) and [Darwin](http://manpagez.com/man/3/realpath/).  The Darwin page states that it will return the "address of newly allocated memory" when the second argument is NULL.  The iOS page does not mention what happens.

I can change it to use NULL if you like.  It looks like the windows implementation in the file already supports that usage.

Reply:
> but I'm unsure if it's supported on our relevant platforms

As I said on #3019 its POSIX 2008, so I would change it to use NULL and see if any platforms report that they are still in the noughties :-)

Reply:
Should the return value be tested and if NULL fallback to preallocating the buffer with max_path?  Or just let it fail if a platform doesn't support this use?

Reply:
On every modern Linux the pathconf value is "unsuitable for allocating memory", so that will fail too if PATH_MAX is not something sane, and IIUC on some systems its not defined.  So I would never use the get_path_max() value to allocate memory, its known to fail on modern systems, whereas realpath() should work on modern systems.  As the system realpath() can legitimately fail for other reasons, allocating memory if it fails is not a good idea.

I don't know if the Windows hack is needed still, lets assume so until some Windows guru says it went out with DOS :)

I would write it as (pseudocode):
```
#if windows and not have_realpath
    gchar *utils_get_real_path(const gchar *file_name){
       the windows implementation allocating PATH_MAX (IIUC it is always defined on Windows)
    }
#else
    gchar *utils_get_real_path(const gchar *file_name){
        return realpath(file_name, NULL);
    }
#endif
```   

We really should check errno after calling `realpath()` since there is a GNU extension that returns a partial path on EACCESS or ENOENT, but hell we almost never check it, why start now :-)

Reply:
For Windows, according to [MS](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd), MAX_PATH = 260 (includes zero terminator).  Longer paths (up to 32767) can be used if certain path prefixes (`\\?\`) are used.  For Win10, long paths can also be enabled with a registry key.  The app must have a manifest file and it applies to only some functions.

Reply:
> Adding the prefix seems like the easiest way to use longer paths on Windows, but changing Windows behavior would be difficult for me to test.

That is probably best left to the future guy that requires longer paths on Windows. Devs tend to checkout code directly to C:\src or similar (possibly *because* of the limitations?) so nobody came up with that requirement so far, to my knowledge.